pkg/services/guardian/guardian.go

```diff
@@ -375,7 +379,12 @@ func (g *dashboardGuardianImpl) getTeams() ([]*team.TeamDTO, error) {
 		return g.teams, nil
 	}
 
-	query := team.GetTeamsByUserQuery{OrgID: g.orgId, UserID: g.user.UserID, SignedInUser: g.user}
+	userIdD, err := identity.IntIdentifier(g.user.GetNamespacedID())
```

Comment:
I think you can avoid this query if it's an API key, otherwise you'll be querying an api key ID as a user ID